crates/oxc_allocator/src/pool.rs

```diff
@@ -0,0 +1,68 @@
+use std::mem::ManuallyDrop;
+use std::sync::{Arc, Mutex};
+
+use crate::Allocator;
+
+/// A thread-safe pool for reusing `Allocator` instances to reduce allocation overhead.
+///
+/// Internally uses a `Vec` protected by a `Mutex` to store available allocators.
+#[derive(Default)]
+pub struct AllocatorPool {
+    items: Arc<Mutex<Vec<Allocator>>>,
+}
+
+/// A guard object representing exclusive access to an `Allocator` from the pool.
+///
+/// On drop, the `Allocator` is reset and returned to the pool.
+pub struct AllocatorPoolGuard {
```

Comment:
Would `AllocatorGuard` be a better name? It guards an `Allocator`, not an `AllocatorPool`. Or `PooledAllocator`?